src/assets/htmx/htmx.md

```diff
@@ -0,0 +1,370 @@
+    hx-post="/save"
+    hx-trigger="submit"
+    hx-target="#todo-list"
+  >
+    <input type="text" name="task" required />
+    <input type="submit" value="save" />
+  </form>
+</div>
+```
+
+Now let's have a look on what is needed on the server side:
+
+```go
+func save(w http.ResponseWriter, r *http.Request) {
+    r.ParseForm()
+    insert(r.FormValue("task")) // save to database
+    taskList := "<div>" + r.FormValue("task") + " </div>"
+    io.WriteString(w, taskList)
+}
+
+func main() {
+    ...
+    http.HandleFunc("/save", save)
+    ...
+}
+```
+![demo](/src/assets/htmx/save.gif)
+
+
+In the HTML above, the form will send a post request to the `/save` endpoint and htmx will replace the content in the div with the id `todo-list`. The response will replace everything in the target element i.e., the element with id `todo-list` will be replaced and looks like this:
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+</div>
+```
+
+After adding another todo item to the list the HTML document looks like this:
+
+```html
+<div id="todo-list">
+  <div>item 2</div>
+</div>
+```
+
+That is not quite what we want. We want to append the new todo item after the last todo item. so we use the attribute `hx-swap` to let htmx know where we want the response to render. The `beforeend` value will append the HTML response before the end of the targeted element.
+
+After the swap the code will look like this:
+
+```html
+<div id="app" class="container">
+  <div id="todo-list"></div>
+  <form
+    id="myForm"
+    hx-post="/save"
+    hx-trigger="submit"
+    hx-target="#todo-list"
+    hx-swap="beforeend"
+  >
+    broadcast<input type="text" name="task" required />
+    <input type="submit" value="save" />
+  </form>
+</div>
+```
+
+Adding first item (item 1):
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+</div>
+```
+
+After adding second item (item 2):
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+  <div>item 2</div>
+</div>
+```
+![demo](/src/assets/htmx/savemultiple.gif)
+
+Adding multiple todo items is working now but on page load (e.g. if we refresh the page) the list of tasks are not being loaded.
+To make that work, we need to make a `GET` request.
+
+```html
+<div id="todo-list" hx-get="/taskList" hx-trigger="load"></div>
+```
+
+In the above HTML, I have added `hx-get` with value `/tasklist` and `hx-trigger` with value 'load'. The `load` value specifies when the request is sent. In this case it's when the HTML file is loaded.
+
+```go
+func listTask(w http.ResponseWriter, r *http.Request) {
+    tastList := getTODO() // gets data from database and returns HTML
+    io.WriteString(w, tastList)
+}
+func main() {
+    ...
+    http.HandleFunc("/save", listTask)
+    ...
+}
+```
+![demo](/src/assets/htmx/getalltask.gif)
+
+Now the list is shown even when the page is refreshed.
+
+For edit let's try something different instead of doing all our operations in a single we will be navigating to another page to do the edit.
```

Comment:
```suggestion
For edit let's try something different: instead of doing all our operations in a single page we will be navigating to another page to do the edit.
```
something is missing after single... single page? single ...?